src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -2010,6 +2127,366 @@ mongoc_client_encryption_rewrap_many_datakey (
    RETURN (ret);
 }
 
+bool
+mongoc_client_encryption_delete_key (
+   mongoc_client_encryption_t *client_encryption,
+   const bson_value_t *keyid,
+   bson_t *reply,
+   bson_error_t *error)
+{
+   bool ret = false;
+   bson_t selector = BSON_INITIALIZER;
+
+   BSON_ASSERT_PARAM (client_encryption);
+   BSON_ASSERT_PARAM (keyid);
+
+   BSON_ASSERT (
+      _coll_has_write_concern_majority (client_encryption->keyvault_coll));
+
+   BSON_ASSERT (keyid->value_type == BSON_TYPE_BINARY);
+   BSON_ASSERT (keyid->value.v_binary.subtype == BSON_SUBTYPE_UUID);
+   BSON_ASSERT (keyid->value.v_binary.data_len > 0u);
+
+   BSON_ASSERT (BSON_APPEND_BINARY (&selector,
+                                    "_id",
+                                    keyid->value.v_binary.subtype,
+                                    keyid->value.v_binary.data,
+                                    keyid->value.v_binary.data_len));
+
+   ret = mongoc_collection_delete_one (
+      client_encryption->keyvault_coll, &selector, NULL, reply, error);
+
+   bson_destroy (&selector);
+
+   RETURN (ret);
```

Comment:
Does it make sense to use tracing macros here if you're not also using `ENTRY`? Perhaps that would be unwanted noise, but I wanted to check in case `RETURN` is always supposed to have a corresponding `ENTRY` line.

This might have been inadvertent, because some of the other methods don't use `RETURN`.

Reply:
Good catch. For consistency with other exported functions, will add appropriate `ENTRY` lines to definitions and ensure proper use of `RETURN`.